src/callbacks_stage/positivity_zhang_shu_dg2d.jl

```diff
@@ -42,4 +42,48 @@ function limiter_zhang_shu!(u, threshold::Real, variable,
 
     return nothing
 end
+
+function limiter_zhang_shu!(u, threshold::Real, variable,
+                            mesh::Union{StructuredMesh{2}, StructuredMeshView{2},
+                                        UnstructuredMesh2D, P4estMesh{2},
+                                        T8codeMesh{2}},
+                            equations, dg::DGSEM, cache)
+    @unpack weights = dg.basis
+
+    @threaded for element in eachelement(dg, cache)
+        # determine minimum value
+        value_min = typemax(eltype(u))
+        for j in eachnode(dg), i in eachnode(dg)
+            u_node = get_node_vars(u, equations, dg, i, j, element)
+            value_min = min(value_min, variable(u_node, equations))
+        end
+
+        # detect if limiting is necessary
+        value_min < threshold || continue
+
+        # compute mean value
+        u_mean = zero(get_node_vars(u, equations, dg, 1, 1, element))
+        total_volume = zero(real(mesh))
+        for j in eachnode(dg), i in eachnode(dg)
+            jacobian_node = abs(inv(cache.elements.inverse_jacobian[i, j, element]))
+            u_node = get_node_vars(u, equations, dg, i, j, element)
+            u_mean += u_node * weights[i] * weights[j] * jacobian_node
+            total_volume += weights[i] * weights[j] * jacobian_node
```

Comment:
```suggestion
            volume_jacobian = abs(inv(cache.elements.inverse_jacobian[i, j, element]))
            u_node = get_node_vars(u, equations, dg, i, j, element)
            u_mean += u_node * weights[i] * weights[j] * volume_jacobian
            total_volume += weights[i] * weights[j] * volume_jacobian
```

I would rename this to be consisted with other places in the code as in [`analysis_dg_2d.jl`](https://github.com/trixi-framework/Trixi.jl/blob/main/src/callbacks_step/analysis_dg2d.jl#L194).

Reply:
Done in [4553ccf](https://github.com/trixi-framework/Trixi.jl/pull/1945/commits/4553ccff2a3189c2e23a3014fbf3ec468c2c8439)